### PR TITLE
Scala 2.13.0-M4

### PR DESCRIPTION
**Pull request by xuwei-k**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -26,7 +43,7 @@ lazy val plugin = build("twotails", "core").settings(
     //"-Ycheck:twotails"
   ),
   libraryDependencies ++= Seq(
-    "org.scalatest" %% "scalatest" % "3.0.0" % "test"
+    "org.scalatest" %% "scalatest" % "3.0.6-SNAP1" % "test"
```
Comment (wheaties) [inline build.sbt:46]: I'm generally not a fan of depending on Snapshot versions. Is this required for M4?

File: core/src/main/scala-2.13-/twotails/Compat.scala [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package twotails
+
+private[twotails] trait Compat { self: SizeLimited =>
+  import self.global._
+
+  // https://github.com/scala/scala/commit/870131bf4b
```
Comment (wheaties) [inline core/src/main/scala-2.13-/twotails/Compat.scala:6]: Thank you. I keep coming back to this every few weeks to bang my head against the NullPointer issues.

File: build.sbt (hunk 1 above)
Comment (xuwei-k) [reply build.sbt:46]: required for Scala 2.13.0-M4. Should we change follows?

```scala
  libraryDependencies += {
    CrossVersion.partialVersion(scalaVersion.value) match {
      case Some((2L, v)) if v >= 13 =>
        "org.scalatest" %% "scalatest" % "3.0.6-SNAP1" % "test"
      case _ =>
        "org.scalatest" %% "scalatest" % "3.0.5" % "test"
    }
  }
```

Comment (wheaties) [reply build.sbt:46]: You know what? This is something we can fix later. I should have given it a think and said "SNAPSHOT for an M# release, why not?"

**Pull request merged**